cmd/commands/subvolume.go

```diff
@@ -40,22 +40,23 @@ var listCmd = &cobra.Command{
 }
 
 var deleteCmd = &cobra.Command{
-	Use:                "delete",
-	Short:              "Deletes a stale subvolume.",
-	DisableFlagParsing: true,
-	Args:               cobra.ExactArgs(3),
-	Example:            "kubectl rook-ceph delete <subvolumes> <filesystem> <subvolumegroup>",
+	Use:     "delete",
+	Short:   "Deletes a stale subvolume.",
+	Args:    cobra.ExactArgs(3),
+	Example: "kubectl rook-ceph delete <subvolumes> <filesystem> <subvolumegroup>",
 	Run: func(cmd *cobra.Command, args []string) {
 		ctx := cmd.Context()
 		subList := args[0]
 		fs := args[1]
 		svg := args[2]
-		subvolume.Delete(ctx, clientSets, operatorNamespace, cephClusterNamespace, subList, fs, svg)
+		retainSnapshots, _ := cmd.Flags().GetBool("retain-snapshots")
+		subvolume.Delete(ctx, clientSets, operatorNamespace, cephClusterNamespace, subList, fs, svg, retainSnapshots)
 	},
 }
 
 func init() {
 	SubvolumeCmd.AddCommand(listCmd)
 	SubvolumeCmd.PersistentFlags().Bool("stale", false, "List only stale subvolumes")
 	SubvolumeCmd.AddCommand(deleteCmd)
+	SubvolumeCmd.PersistentFlags().Bool("retain-snapshots", false, "Delete subvolume but retain the snapshot")
```

Comment:
This should be the default behaviour, we dont need to have a new flag from user, because you can never delete a subvolume which is having snapshots without `retain-snapshot` flag in ceph CLI.

Reply:
The idea was to list the subvolmes along with the status(if it has snapshots or not), so user can pass this flag and delete the stale subvolumes, even if it has snapshots.

Reply:
IMHO we dont need any flag this should be the default behaviour